src/nl/hannahsten/texifyidea/inspections/latex/codestyle/LatexUsePackageInPackageInspection.kt

```diff
@@ -1,20 +1,38 @@
 package nl.hannahsten.texifyidea.inspections.latex.codestyle
 
+import com.intellij.codeInspection.InspectionManager
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.psi.PsiFile
 import nl.hannahsten.texifyidea.file.ClassFileType
 import nl.hannahsten.texifyidea.file.StyleFileType
-import nl.hannahsten.texifyidea.inspections.TexifyRegexInspection
-import java.util.regex.Pattern
+import nl.hannahsten.texifyidea.index.DefinitionBundle
+import nl.hannahsten.texifyidea.inspections.AbstractTexifyCommandBasedInspection
+import nl.hannahsten.texifyidea.lang.LContextSet
+import nl.hannahsten.texifyidea.psi.LatexCommands
+import nl.hannahsten.texifyidea.psi.nameWithoutSlash
 
-class LatexUsePackageInPackageInspection : TexifyRegexInspection(
-    inspectionDisplayName = "Use of \\usepackage{...} instead of \\RequirePackage{...}",
-    inspectionId = "UsePackageInPackage",
-    errorMessage = { "Use \\RequirePackage{...} instead of \\usepackage{...}" },
-    // The \\usepackage syntax is \\usepackage[<options>]{<package name>}[<version info>].
-    pattern = Pattern.compile("\\\\usepackage(\\[[^]]*])?\\{([^}]*)}(\\[[^]]*])?"),
-    quickFixName = { "Replace with \\RequirePackage" },
-    replacement = { matcher, _ -> "\\RequirePackage${matcher.group(1) ?: ""}{${matcher.group(2)}}${matcher.group(3) ?: ""}" },
-    cancelIf = { _, psiFile ->
-        // Cancel if the usepackage was found outside a class or style file.
-        psiFile.virtualFile?.extension !in setOf(ClassFileType.defaultExtension, StyleFileType.defaultExtension)
+class LatexUsePackageInPackageInspection : AbstractTexifyCommandBasedInspection(
+    inspectionId = "UsePackageInPackage"
+) {
+    private val applicableFileExtensions = setOf(ClassFileType, StyleFileType)
+
+    override fun isAvailableForFile(file: PsiFile): Boolean {
+        val fileType = file.virtualFile?.fileType ?: file.fileType
+        // don't know why but sometimes file.fileType is not the same as file.virtualFile?.fileType
```

Comment:
I can reproduce the issue, I saw a `.sty` file where `file.fileType` is incorrectly LatexFileType (tex) but `file.virtualFile.fileType` is StyleFileType. Very strange